lib/include/openturns/OTtestcode.hxx

```diff
@@ -454,6 +429,596 @@ void assert_equal(const T & a, const T & b, const String errMsg = "")
+
+    Distribution standardRep(distribution_.getStandardRepresentative());
+    LOGTRACE(OSS() << "Standard representative=" << standardRep);
+
+    if (distribution_.isDiscrete() && (distribution_.getDimension() == 1))
+    {
+      const Complex z(0.3, 0.7);
+      Complex GF = distribution_.computeGeneratingFunction(z);
+      LOGTRACE(OSS() << "generating function=" << GF);
+      Complex LGF = distribution_.computeLogGeneratingFunction(z);
+      LOGTRACE(OSS() << "log generating function=" << LGF);
+    }
+  }
+
+  void checkCharacteristicFunction() const
+  {
+    if (distribution_.getDimension() == 1)
+    {
+      LOGTRACE(OSS() << "checking characteristic function...");
+      const Scalar t = 1.0;
+      Complex CF = distribution_.computeCharacteristicFunction(t);
+      LOGTRACE(OSS() << "characteristic function=" << CF);
+      Complex LCF = distribution_.computeLogCharacteristicFunction(t);
+      LOGTRACE(OSS() << "log characteristic function=" << LCF);
+    }
+  }
+
+  void checkComparison() const
+  {
+    LOGTRACE(OSS() << "checking comparison operators...");
+    if (!(distribution_ == distribution_))
+      throw TestFailed(OSS() << "operator==(self) failed for " << distribution_);
+    if (distribution_ != distribution_)
+      throw TestFailed(OSS() << "operator==(self) failed for " << distribution_);
+    const Distribution distribution2(distribution_);
+    if (!(distribution2 == distribution_))
+      throw TestFailed(OSS() << "operator==(copy) failed for " << distribution_);
+    if (distribution2 != distribution_)
+      throw TestFailed(OSS() << "operator!=(copy) failed for " << distribution_);
+    const Dirac dirac(42.0);
+    if (distribution_ == dirac)
+      throw TestFailed(OSS() << "operator==(other) failed for " << distribution_);
+    if (!(distribution2 != dirac))
+      throw TestFailed(OSS() << "operator!=(other) failed for " << distribution_);
+  }
+  
+  void checkPDF() const
+  {
+    LOGTRACE(OSS() << "checking PDF...");
+    const Sample sample(distribution_.getSample(10));
+    for (UnsignedInteger i = 0; i < sample.getSize(); ++ i)
+    {
+      const Point x(sample[i]);
+      const Scalar pdf = distribution_.computePDF(x);
+      LOGTRACE(OSS() << "pdf=" << pdf);
+      if (!(pdf > 0.0))
+        throw TestFailed(OSS() << "pdf(x) failed for " << distribution_);
+    }
+  }
+
+  void checkLogPDF() const
+  {
+    LOGTRACE(OSS() << "checking LogPDF...");
+    const Sample sample(distribution_.getSample(10));
+    for (UnsignedInteger i = 0; i < sample.getSize(); ++ i)
+    {
+      const Point x(sample[i]);
+      const Scalar lpdf2 = std::log(distribution_.computePDF(x));
+      LOGTRACE(OSS() << "log(pdf)=" << lpdf2);
+      const Scalar lpdf1 = distribution_.computeLogPDF(x);
+      LOGTRACE(OSS() << "logpdf  =" << lpdf1);
+      assert_almost_equal(lpdf1, lpdf2, 1e-5, 1e-8, distribution_.__repr__());
+    }
+  }
+
+  void checkCDF() const
```

Comment:
There should be a `checkComplementaryCDF` method to check `computeComplementaryCDF`.